### ISSUE TITLE
6$: Improve error recovery for missing } in parameter destructure

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#1063]
```javascript
class AnalyticsDB {
  static newInMemory() {
  }

  addDownload({timestamp, url, downloaderIP, downloaderUserAgent) {
  }
}
```
Expected: missing `}` to close destructuring thingy
Observed: E0054: unexpected token

**Comment by yashmasani** [comment]
Hi, I would like to work on this

**Comment by strager** [comment]
@yashmasani Sure!

**Comment by strager** [comment]
@yashmasani Either message is good.
